### DISCUSSION TITLE
Why is validation performed first, and then checking for an authorized user?

### DISCUSSION THREAD

**Discussion by zabavnikov** [rebing/graphql-laravel#935] [Q&A] [answered]
Mutation:

![image](https://user-images.githubusercontent.com/8766240/182760416-8cac06cc-fdd8-48dd-8880-758891d81b7f.png)

I run the mutation as a guest and expect the authorization method to fire first, but the validation is executed first.

![image](https://user-images.githubusercontent.com/8766240/182760671-654faa0c-2bcb-43a0-91eb-f349ea2d8e7d.png)
![image](https://user-images.githubusercontent.com/8766240/182760707-5edc0528-f315-40e3-b5d7-c7ee2f239733.png)

I fill in all required fields

![image](https://user-images.githubusercontent.com/8766240/182760831-ed69f643-e6aa-4e4b-91b2-65172082797b.png)
![image](https://user-images.githubusercontent.com/8766240/182760854-3803f0c4-8950-4b08-8d48-f719c2c26968.png)

**Comment by zabavnikov** [comment, marked as answer]
https://github.com/rebing/graphql-laravel/issues/481

**Reply by mfn** [reply]
Hi,

yes, basically it's a forth and back and there were never clear arguments pro/con which one is better and now we settled on the current one and I think it's unlikely to change.

If you need full control, don't use the built-in helper methods and just to it in your own in `resolve()`.
